shared-libs/transitions/src/transitions/registration.js

```diff
@@ -325,50 +350,170 @@ const setId = (options) => {
+  const patientShortcode = options.doc.patient_id;
+  const patientNameField = getPatientNameField(options.params);
+
+  // create a new patient with this patient_id
+  const patient = {
+    name: doc.fields[patientNameField],
+    reported_date: doc.reported_date,
+    patient_id: patientShortcode,
+    source_id: doc._id,
+  };
+
+  if (options.params.contact_type) {
+    patient.type = 'contact';
+    patient.contact_type = options.params.contact_type;
+  } else {
+    patient.type = 'person';
+  }
 
   return utils
-    .getPatientContactUuid(patientShortcode)
+    .getContactUuid(patientShortcode)
     .then(patientContactId => {
       if (patientContactId) {
+        doc.patient_id = patientShortcode;
         return;
       }
 
-      return db.medic
-        .query('medic-client/contacts_by_phone', { key: doc.from, include_docs: true })
-        .then(result => {
-          const contact = _.result(_.first(result.rows), 'doc');
-          lineage.minify(contact);
-          // create a new patient with this patient_id
-          const patient = {
-            name: doc.fields[patientNameField],
-            created_by: contact && contact._id,
-            parent: contact && contact.parent,
-            reported_date: doc.reported_date,
-            patient_id: patientShortcode,
-            source_id: doc._id,
-          };
-          if (options.params.contact_type) {
-            patient.type = 'contact';
-            patient.contact_type = options.params.contact_type;
-          } else {
-            patient.type = 'person';
-          }
-          // include the DOB if it was generated on report
-          if (doc.birth_date) {
-            patient.date_of_birth = doc.birth_date;
-          }
-          return db.medic.post(patient);
-        });
+      return getParent(options).then(parent => {
+        if (!parent) {
+          return;
+        }
+
+        if (!isValidParent(parent, patient)) {
+          transitionUtils.addRejectionMessage(
+            options.doc,
+            options.registrationConfig,
+            PARENT_INVALID,
+            { templateContext: { parent } }
+            );
+          return;
+        }
+
+        // include the DOB if it was generated on report
+        if (doc.birth_date) {
+          patient.date_of_birth = doc.birth_date;
+        }
+
+        // assign patient in doc with full parent doc - to be used in messages
+        doc.patient = Object.assign({ parent }, patient);
+        patient.parent = lineage.minifyLineage(parent);
+        patient.created_by = doc.contact && doc.contact._id;
+
+        return db.medic.post(patient);
+      });
     });
+};
+
+const addPlace = (options) => {
+  const doc = options.doc;
+  const placeShortcode = doc.place_id;
+  const placeNameField = getPlaceNameField(options.params);
+
+  // create a new place with this place_id
+  const place = {
+    name: doc.fields[placeNameField],
+    type: 'contact',
+    contact_type: options.params.contact_type,
```

Comment:
Does this mean we only support creating place by SMS for configurable hierarchy configs?

Reply:
I've changed it so it will use the old format for hardcoded contact types.